File: java/cours/src/session3/TD/MotRecursif/FabriqueMot.java
package session3.TD.MotRecursif;

public interface FabriqueMot {
	Mot createInduct(char lettre, Mot next);
	Mot createFromConcat(Mot m1, Mot m2);
	Mot createVide();
	Mot createFromString(String mot);
}

File: typescript/session3/MotRecursif/FabriqueMot.ts
import { Char } from '../Char';
import { Mot } from './mot';

export interface FabriqueMot {
  createInduct(lettre: Char, next: Mot): Mot;
  createFromConcat(m1: Mot, m2: Mot): Mot;
  createVide(): Mot;
  createFromString(mot: string): Mot;
}